code/modules/events/immovablerod.dm

```diff
@@ -101,9 +103,23 @@
 		qdel(src)
 		return
 
-	if(loc.density)
-		loc.ex_act(2)
-		if(prob(25))
+	if(loc && !istype(loc,/turf/space))
+		if(fuckMyShitUp)
+			loc.ex_act(1)
+		else if(loc.density)
+			if(istype(loc,/turf/simulated/wall/r_wall))
+				loc.ex_act(1)
```

Comment:
Well we need a general destruction proc. It doesn't help that a lot of things define `destroy()` for that as well.